Modules/m68k-amigaos-gcc10/CMakeASM_VASMInformation.cmake

```diff
@@ -1,10 +1,20 @@
-SET(ASM_DIALECT "_VASM")
-SET(CMAKE_ASM_VASM_SOURCE_FILE_EXTENSIONS vasm;asm)
+# Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
+# file Copyright.txt or https://cmake.org/licensing for details.
 
-IF(NOT CMAKE_ASM_VASM_COMPILE_OBJECT)
-	SET(CMAKE_ASM_VASM_COMPILE_OBJECT "<CMAKE_ASM_VASM_COMPILER> <FLAGS> -o <OBJECT> <SOURCE>")
-ENDIF(NOT CMAKE_ASM_VASM_COMPILE_OBJECT)
+# Support for vasm assembler: m68k, ppc, et al.
 
-INCLUDE(CMakeASMInformation)
-SET(ASM_DIALECT)
+set(CMAKE_ASM_VASM_SOURCE_FILE_EXTENSIONS vasm;asm)
+
+# Allow overriding object format: ELF is usually the best option but e.g. m68k-amigaos might expect hunk
+if(NOT CMAKE_ASM_VASM_OBJECT_FORMAT)
+  set(CMAKE_ASM_VASM_OBJECT_FORMAT elf)
+endif()
+
+if(NOT CMAKE_ASM_VASM_COMPILE_OBJECT)
+  set(CMAKE_ASM_VASM_COMPILE_OBJECT "<CMAKE_ASM_VASM_COMPILER> <FLAGS> -F${CMAKE_ASM_VASM_OBJECT_FORMAT} -o <OBJECT> <SOURCE>" CACHE STRING "" FORCE)
```

Comment:
dunno why but I had to add `CACHE STRING "" FORCE` here - NASM config doesn't have that part and probably works correctly 😕

Reply:
Perhaps a question to ask the CMake devs